parsl/executors/status_handling.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import threading
+from abc import abstractmethod
+from typing import List, Any, Dict
+
+from parsl.providers.provider_base import JobStatus, ExecutionProvider, JobState
+
+
+logger = logging.getLogger(__name__)
+
+
+class StatusHandlingMixin(object):
+    tasks = None
+    provider = None
+
+    def __init__(self, provider=None):
+        super().__init__()
+        self._provider = provider
+        # errors can happen during the sumbit call to the provider; this is used
+        # to keep track of such errors so that they can be handled in one place
+        # together with errors reported by status()
+        self._simulated_status = {}
```

Comment:
_job_fail_status would be misleading, no? It would suggest that it contains the status of all failed jobs. Perhaps _simulated_failures. Or maybe it doesn't matter, it being clearly documented in the comment above.